grails-app/services/org/pih/warehouse/core/DocumentService.groovy

```diff
@@ -1571,5 +1573,31 @@ class DocumentService {
         }
     }
 
+    List<Document> getAllDocumentsBySupplierOrganization(Organization supplierOrganization) {
+        List<Order> orders = Order.createCriteria().list {
+            eq("originParty", supplierOrganization)
+            eq("orderTypeCode", OrderTypeCode.PURCHASE_ORDER)
+        }
+
+        def documents = []
+
+        orders.each { order ->
+            order.documents.collect { document ->
```

Comment:
This should be ok for a first pass, but did you try to get it to work with projections?

Reply:
Here's what it might look like
```
    List<Document> getAllDocumentsBySupplierOrganization(Organization supplierOrganization) {
        return Order.createCriteria().list {
            resultTransformer(CriteriaSpecification.ALIAS_TO_ENTITY_MAP)
            projections {
                property("id", "orderId")
                property("orderNumber", "orderNumber")
                property("name", "orderDescription")
                origin {
                    property("name", "origin")
                }
                destination {
                    property("name", "destination")
                }
                documents {
                    property("id", "documentId")
                    property("documentNumber", "documentNumber")
                    documentType {
                        property("name", "documentType")
                    }
                    property("name", "documentName")
                    property("contentType", "fileType")
                    property("fileUri", "fileUri")
                }
            }
            eq("originParty", supplierOrganization)
            eq("orderTypeCode", OrderTypeCode.PURCHASE_ORDER)
        }
    }
```

Reply:
And here's what the data looks like
![image](https://user-images.githubusercontent.com/1162935/119865981-e68f4480-bee1-11eb-9a4e-630053f39c72.png)